HPA Specificity Improvements

DirectXMan12 opened the pull request:
This commit describes the HPA changes that went into 1.11, including
the addition of label selectors and the rework of the API.

cc kubernetes/features#117
cc kubernetes/community#2055
cc kubernetes/kubernetes#64097

DirectXMan12 commented:
/hold
this is a placeholder as per the release requirements.  Will have content shortly.

mistyhacks commented:
Thanks, just remove the hold when you're ready for review.

zparnold commented:
@DirectXMan12 Hey there! Looks like we still need some docs to get this feature ready for release

Can you let me know when that’ll happen please? If there’s anything I can do to assist please let me know

DirectXMan12 commented:
/unhold

mistyhacks commented:
/hold cancel

👼

mistyhacks commented:
@DirectXMan12 I've made two copyedit commits (one per file, for ease of review). PTAL and LGTM if good.

/approve

DirectXMan12 commented on content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md:
```diff
@@ -202,8 +202,8 @@ the only other supported resource metric is memory.  These resources do not chan
-requested value.  To do so, use the `target` type of `AverageValue`, and thus set `target.averageValue`
-field instead of the `target.averageUtilization` field.
+requested value, by using the `target` type of `AverageValue` instead of `averageUtilization`. This sets the
+`target.averageValue` field instead of the `target.averageUtilization` field.
```
this is wrong -- it's a discriminated/tagged union (`target` is the tag, `averageUtilization` and friends are the value).  You need to set `target` to `AverageValue` *and then* set `target.averageValue` yourself.  This wording implies the latter gets set automatically.  The original intention was

> if you set `target` to `Xyz`, you must also set the `xyz` field

DirectXMan12 commented on content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md:
```diff
@@ -225,12 +225,13 @@ pods:
-by the number of pods before being compared to the target.  Object metrics look like this:
+The second alternative metric type is *object metrics*. These metrics describe a different
+object in the same namespace, instead of describing pods. The metrics are not
+fetched from the object, but only describe it. Object metrics support `target` types of
```
IMO, the `but` sounds weird here.  I'd use

> the metrics are not fetched from the object; they only describe it

or

> the metrics describe the object, but are not necessarily fetched from it

DirectXMan12 commented:
added some comments on the copyedit

mistyhacks commented:
I'm confused about the current state of this, because you commented on my edits, but your comments are now collapsed, but I don't think they were addressed. Can you let me know? Also, the Hugo shortcode thing -- I thought I fixed that and it is back wrong again. Did you force-push over my commit?

mistyhacks commented:
It looks like a bit of a mess and I think some of it would be fixed by rebasing again on `release-1.11`.

DirectXMan12 commented:
I didn't push anything new.  The corresponding PR didn't make 1.11 due to a couple hiccups, so we'll have to rebase on 1.12 when ready ;-)

mistyhacks commented:
/hold

zacharysarah commented:
@DirectXMan12 👋 Just an early heads up that merge conflicts need to be resolved before this PR can merge.

DirectXMan12 commented:
I'm force-pushing this, because it looks like a merge commit was added to this PR, which has seriously messed with things.  I've cherry-picked in misty's copyedits, added in my comments on the copyedits, and squashed them back down.

mistyhacks commented:
/lgtm

If it's no longer WIP, don't forget to change the title.

/hold
(for @zparnold )

zparnold commented:
/hold cancel
/approve
